edb/schema/functions.py

```diff
@@ -2440,3 +2443,103 @@ def compile_function(
         )
 
     return compiled
+
+
+def compile_function_inline(
+    schema: s_schema.Schema,
+    context: sd.CommandContext,
+    *,
+    body: s_expr.Expression,
+    func_name: sn.QualName,
+    params: FuncParameterList,
+    language: qlast.Language,
+    return_type: s_types.Type,
+    return_typemod: ft.TypeModifier,
+    track_schema_ref_exprs: bool=False,
+    inlining_context: Optional[qlcontext.ContextLevel],
+) -> irast.Set:
+    """Compile a function body to be inlined."""
+    assert language is qlast.Language.EdgeQL
+
+    from edb.edgeql.compiler import dispatch
+    from edb.edgeql.compiler import inference
+    from edb.edgeql.compiler import pathctx
+    from edb.edgeql.compiler import setgen
+    from edb.edgeql.compiler import stmtctx
+
+    ctx = stmtctx.init_context(
+        schema=schema,
+        options=get_compiler_options(
+            schema,
+            context,
+            func_name=func_name,
+            params=params,
+            track_schema_ref_exprs=track_schema_ref_exprs,
+            inlining_context=inlining_context,
+        ),
+        inlining_context=inlining_context,
+    )
```

Comment:
Do we really need to go through `init_context` here? I think we probably should be trying to share the context mostly. (I think it probably would work if we create a `detached()` context?)

Reply:
I tried to see if I could get a detached context working but I think the inlined context needs to have more than what is available in a detached context.